modules/orionode/index.js

```diff
@@ -135,42 +246,23 @@ function startServer(options) {
 		passport.initialize(),
 		passport.session()
 	].concat(options.authenticate || []);
-	
-	loadEndpoints(false);
-	let CloudFoundry = require('./lib/cf').CloudFoundry;
-	app.use('/metrics', require('./lib/metrics').router(options));
-	app.use('/version', require('./lib/version').router(options));
-	let About = require('./lib/about').About;
-	app.use('/about', new About().createRouter(options));
-	loadEndpoints(true);
-	app.use(require('./lib/user').router(options));
-	app.use('/site', options.authenticate, checkAuthenticated, checkAccessRights, require('./lib/sites')(options));
-	app.use('/task', options.authenticate, checkAuthenticated, require('./lib/tasks').router(options));
-	app.use('/filesearch', options.authenticate, checkAuthenticated, require('./lib/search')(options));
-	app.use('/file*', options.authenticate, checkAuthenticated, checkAccessRights, require('./lib/file')(options));
-	app.use('/workspace*', options.authenticate, checkAuthenticated, checkAccessRights, require('./lib/workspace')(options));
-	app.use('/gitapi', options.authenticate, checkAuthenticated, require('./lib/git')(options));
-	app.use('/cfapi', options.authenticate, checkAuthenticated, new CloudFoundry().createRouter(options));
-	app.use('/prefs', options.authenticate, checkAuthenticated, require('./lib/prefs').router(options));
-	app.use('/xfer', options.authenticate, checkAuthenticated, require('./lib/xfer').router(options));
-	if (options.configParams.get("isElectron")) {
-		app.use('/update', options.authenticate, checkAuthenticated, require('./lib/update').router(options));
+	let serverconf = options.configParams.get("orion.server.config") ? require(options.configParams.get("orion.server.config")) : [];
+	if(serverconf && serverconf.endpoints) {
+		loadEndpoints(serverconf.endpoints, options);
 	}
-	if(options.configParams.get("orion.collab.enabled")) {
-		app.use('/sharedWorkspace', options.authenticate, checkAuthenticated, require('./lib/sharedWorkspace').router(options));
+	let io = socketio.listen(options.server, { 'log level': 1, path: (listenContextPath ? contextPath : '' ) + '/socket.io' });
+	if(serverconf && Array.isArray(serverconf.languages)) {
+		loadLanguageServers(serverconf.languages, io);
 	}
-	
-	var io = socketio.listen(options.server, { 'log level': 1, path: (listenContextPath ? contextPath : '' ) + '/socket.io' });
 	ttyShell.install(options, io);
```

Comment:
Can we optionally install ttyShell?

Reply:
I opened a separate issue to track that - it will take a bit of thinking how to generalize / refactor since it needs socket + an 'install':

https://bugs.eclipse.org/bugs/show_bug.cgi?id=530843